daemon/logger/loggerutils/logfile.go

```diff
@@ -641,6 +642,15 @@ func followLogs(f *os.File, logWatcher *logger.LogWatcher, notifyRotate chan int
 }
 
 func watchFile(name string) (filenotify.FileWatcher, error) {
+
+	// FileWatcher on Windows files is based on the syscall notifications which has an issue becuase of file caching.
+	// It is based on ReadDirectoryChangesW() which doesn't detect writes to the cache. It detects writes to disk only.
+	// Becuase of the OS lazy writing, we don't get notifications for file writes and thereby the watcher
+	// doesn't work. Hence for Windows we will use poll based notifier.
+	if runtime.GOOS == "windows" {
+		return watchFileForWindows(name)
```

Comment:
This ends up duplicating a lot.
Can we set `fileWatcher` to a polling watcher for Windows and keep using the rest of this function?